netket/experimental/operator/_fermion_operator_2nd.py

```diff
@@ -231,9 +238,33 @@ def from_openfermion(
     def __repr__(self):
         return (
             f"FermionOperator2nd(hilbert={self.hilbert}, "
-            f"n_operators={len(self._operators)}, dtype={self.dtype})"
+            f"n_operators={len(self.terms)}, dtype={self.dtype})"
         )
 
+    @property
+    def terms(self) -> OperatorTermsList:
+        return self._terms
+
+    @property
+    def weights(self) -> OperatorWeightsList:
+        return self._weights
+
+    @property
+    def constant(self) -> Union[float, complex]:
+        return self._constant
+
+    @property
+    def operator_dict(self) -> OperatorDict:
+        """
+        Return a dict with terms as keys and weights as values.
+        This does not include the constant term.
+        """
+        return dict(zip(self.terms, self.weights))
```

Comment:
Oh nice. We're moving to a tuple of lists. This should be quite faster at manipulations...
Then why do we need this?

Reply:
Some of the functionality uses this to add duplicates of terms together, for example. Plus, it's just to provide a bit of backward compatibility.

Reply:
If I am not mistaken there is only a single internal use in `__iadd__`, plus once again in a test.

I am entirely ok leaving this property there (with a deprecation warning) to change behaviour more smoothly, but would like to get rid of the dictionary manipulations because this is what is needed to address the slowdowns of #1576 . You could do the same thing you did for Pauli strings.

Reply:
How do you propose is the best way to merge duplicates and remove zeros?

Reply:
Could you maybe run the same benchmarks like you did before, to see if this helps already? I would still need to find a fast way to merge duplicates. It's not as simple as in the PauliStrings case, since the keys are tuple trees...

Reply:
Yeah, I was afraid so. I've just pushed a version that does not prune all the time, but it just adds a "reduce" function... I call it in construction and setup now.

Reply:
I recently experimented with symbolic manipulation with very large expressions, and actually the trick that did it was to remove zeros as they come when adding two dictionaries..

Here `OpSentence` is the equivalent of your operator, and `A.values` is the equivalent of your `operator_dict`. This is very fast and required no further optimisations.

```python
@dispatch
def __add__(A: OpSentence, B: OpSentence):
    res = A.values.copy()
    for k, v in B.values.items():
        sv = res.get(k, None)
        if sv is None and v != 0:
            res[k] = v
        elif sv is not None:
            v = sv + v
            if v == 0:
                del res[k]
            else:
                res[k] = v
    return OpSentence(res)
```

Reply:
Ok, I did it this way now, take a look. I do keep the terms and weights list though, so I need to make a list of keys and values in the end.